bundles/org.openhab.binding.benqprojector/src/main/java/org/openhab/binding/benqprojector/internal/discovery/BenqProjectorDiscoveryService.java

```diff
@@ -0,0 +1,159 @@
+ * for discovering BenQ projectors using the AMX Device Discovery protocol.
+ *
+ * @author Mark Hilbush - Initial contribution
+ * @author Michael Lobstein - Adapted for the BenQ Projector binding
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.benqprojector")
+public class BenqProjectorDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(BenqProjectorDiscoveryService.class);
+    private @Nullable ScheduledFuture<?> benqDiscoveryJob;
+
+    // Discovery parameters
+    public static final boolean BACKGROUND_DISCOVERY_ENABLED = true;
+    public static final int BACKGROUND_DISCOVERY_DELAY_TIMEOUT_SEC = 10;
+
+    private NetworkAddressService networkAddressService;
+    private final TranslationProvider translationProvider;
+    private final LocaleProvider localeProvider;
+    private final @Nullable Bundle bundle;
+
+    private boolean terminate = false;
+
+    @Activate
+    public BenqProjectorDiscoveryService(@Reference NetworkAddressService networkAddressService,
+            @Reference TranslationProvider translationProvider, @Reference LocaleProvider localeProvider) {
+        super(SUPPORTED_THING_TYPES_UIDS, 0, BACKGROUND_DISCOVERY_ENABLED);
+        this.networkAddressService = networkAddressService;
+        this.translationProvider = translationProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(BenqProjectorDiscoveryService.class);
+
+        benqDiscoveryJob = null;
+        terminate = false;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (benqDiscoveryJob == null) {
+            terminate = false;
+            logger.debug("Starting background discovery job in {} seconds", BACKGROUND_DISCOVERY_DELAY_TIMEOUT_SEC);
+            benqDiscoveryJob = scheduler.schedule(this::discover, BACKGROUND_DISCOVERY_DELAY_TIMEOUT_SEC,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        ScheduledFuture<?> benqDiscoveryJob = this.benqDiscoveryJob;
+        if (benqDiscoveryJob != null) {
+            terminate = true;
+            benqDiscoveryJob.cancel(false);
+            this.benqDiscoveryJob = null;
+        }
+    }
+
+    @Override
+    public void startScan() {
+    }
+
+    @Override
+    public void stopScan() {
+    }
+
+    private synchronized void discover() {
+        logger.debug("Discovery job is running");
+        MulticastListener benqMulticastListener;
+        String local = "127.0.0.1";
+
+        try {
+            String ip = networkAddressService.getPrimaryIpv4HostAddress();
+            benqMulticastListener = new MulticastListener((ip != null ? ip : local));
+        } catch (SocketException se) {
```

Comment:
`SocketException` is a subclass of `IOException`, so is there any need to catch this explicitly when handled exactly the same way as below? Only difference is the logged text "Socket" vs. "IO", but the message should be sufficient.

Reply:
Removed the extra catch block.